reference/tools/tiup: add manage-component doc (#2472)
pingcap/docs: reference/tools/tiup/manage-component.md

---

> title: Use TiUP to Manage Components

Reviewer: ```suggestion
title: Manage TiUP Components with TiUP commands
```

> summary: Learn how to manage components using TiUP.

Reviewer: Need to update accordingly.

category: tools
---

> # Use TiUP to Manage Components

Reviewer: ```suggestion
# Manage TiUP Components with TiUP commands
```

> You can use the following TiUP sub-commands to manage components:

Reviewer: ```suggestion
You can use the following TiUP commands to manage components in the TiUP ecosystem:
```
Refer to https://github.com/pingcap/docs/pull/2482/files.

> - list: Queries the component list. By using this sub-command, you can see all the optional components to install and all the optional versions of each component.

Reviewer: ```suggestion
- list: Queries the component list. By using this command, you can see all the optional components to install and all the optional versions of each component.
```
I think we can safely use "command" instead of "sub-command" because the meanings are both clear here. But to keep the term usage unified, I suggest using "(TiUP) command". What do you think? @TomShawn

Author: Good idea.

- install: Installs the specific version of a component.
- update: Updates a component to the latest version.
- uninstall: Uninstalls a component.
- status: Checks the status of a running component.
- clean: Cleans up the instance on which a component is deployed.
- help: Prints the help information. If the sub-command is appended to a command, the usage of this command is printed. 

> This document introduces the common component management operations and the corresponding sub-commands.

Reviewer: ```suggestion
This document introduces the common component management operations and the corresponding TiUP commands.
```

- `--refresh`: gets the latest list of components on the server and the version list of each component.

Example 1: View all currently installed components.

{{< copyable "shell-regular" >}}

```shell
tiup list --installed
```

Example 2: Get a list of the TiKV component of all installable versions from the server.

{{< copyable "shell-regular" >}}

```shell
tiup list tikv --refresh
```

## Install components

You can use the `tiup install` command to query the component list. This usage of this command is as follows:

- `tiup install <component>`: installs the latest stable version of a specified component.
- `tiup install <component>:[version]`: installs the specified version of a specified component.

Example 1: Use TiUP to install the latest stable version of TiDB.

{{< copyable "shell-regular" >}}

```shell
tiup install tidb
```

Example 2: Use TiUP to install the nightly version of TiDB.

{{< copyable "shell-regular" >}}

```shell
tiup install tidb:nightly
```

Example 3: Use TiUP to install TiKV v3.0.6.

{{< copyable "shell-regular" >}}

```shell
tiup install tikv:v3.0.6
```

## Upgrade components

> After PingCAP has released a new version of a component, you can use the `tiup update` command to upgrade this component. The usage of this command is basically the same as that of `tiup install`, except for the following flags:

Reviewer: ```suggestion
After a new version of a component is published, you can use the `tiup update` command to upgrade this component. The usage of this command is basically the same as that of `tiup install`, except for the following flags:
```
The owner of a TiUP component is not necessarily PingCAP. The TiUP ecosystem is open-sourced.

```shell
tiup clean --all
```

### Uninstall components

The components installed using TiUP take up local disk space. If you do not want to keep too many components of old versions, you can check which versions of a component are currently installed, and then uninstall this component.

You can use the `tiup uninstall` command to uninstall all versions or specific versions of a component. This command also supports uninstalling all components. The command usage is as follows:

{{< copyable "shell-regular" >}}

```bash
tiup uninstall [component][:version] [flags]
```

The following flags are supported in this command:

- `--all`: Uninstalls all components or versions.
- `--self`: Uninstalls TiUP itself.

`component` is the component to be uninstalled. `version` is the version to be uninstalled. Both `component` and `version` can be ignored in the `tiup uninstall` command. If you ignore either one of these two, you need to add the `--all` flag.

- If the version is ignored, adding `--all` means to uninstall all versions of this component.
- If the version and the component are both ignored, adding `--all` means to uninstall all components of all versions.

> Example: Uninstall TiDB v3.0.8.

Reviewer 2: ```suggestion
Example 1: Uninstall TiDB v3.0.8.
```

{{< copyable "shell-regular" >}}

```shell
tiup uninstall tidb:v3.0.8
```

> Example: Uninstall TiKV of all versions.

Reviewer 2: ```suggestion
Example 2: Uninstall TiKV of all versions.
```

{{< copyable "shell-regular" >}}

```shell
tiup uninstall tikv --all
```

> Example: Uninstall all installed components.

Reviewer 2: ```suggestion
Example 3: Uninstall all installed components.
```
